internal/clients/provider_config.go

```diff
@@ -264,9 +264,14 @@ func UseProviderSecret(ctx context.Context, data []byte, profile, region string)
 // AssumeRoleWithWebIdentity & AssumeRoles.
 func GetRoleChainConfig(ctx context.Context, pcs *v1beta1.ProviderConfigSpec, cfg *aws.Config) (*aws.Config, error) {
 	pCfg := cfg
+	regionOpt := func(o *sts.Options) {
+		if pcs.AssumeRoleSTSRegion != nil {
+			o.Region = *pcs.AssumeRoleSTSRegion
+		}
+	}
```

Comment:
@ctkeyser @ulucinar I wonder if it would make sense to fall back to the AWS global partition here if an STS region is not defined? The AWS SDK itself [will do so](https://github.com/aws/aws-sdk-go-v2/issues/1778) if it does not recognize the region that is passed, but will error if no region is passed. Thoughts?

Reply:
Hi @hasheddan,
Makes sense to me.

Reply:
that sounds good to me, but I'm not sure what that change looks like. I tried changing region to `global` and `aws-global` and I ended up getting errors saying dns doesn't resolve for `sts.aws-global.amazonaws.com` or `sts.global.amazonaws.com`. dns does resolve for `sts.us-west-1.amazonaws.com` though

I also don't see STS in this list of global endpoints https://docs.aws.amazon.com/general/latest/gr/rande.html#global-endpoints

Reply:
@ctkeyser I believe it may just be `sts.amazonaws.com` https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_temp_enable-regions.html

Reply:
ok I tested just falling back to `aws-global` if no region set, and that does work in my environment:
```diff
--- a/internal/clients/provider_config.go
+++ b/internal/clients/provider_config.go
@@ -265,8 +265,8 @@ func UseProviderSecret(ctx context.Context, data []byte, profile, region string)
 func GetRoleChainConfig(ctx context.Context, pcs *v1beta1.ProviderConfigSpec, cfg *aws.Config) (*aws.Config, error) {
        pCfg := cfg
        regionOpt := func(o *sts.Options) {
-               if pcs.AssumeRoleSTSRegion != nil {
-                       o.Region = *pcs.AssumeRoleSTSRegion
+               if cfg.Region == "" {
+                       o.Region = GlobalRegion
                }
        }
        for _, aro := range pcs.AssumeRoleChain {
```
I was able to assume a role and create an IAM resource with that
I also tried changing getRegion to return GlobalRegion if no region was found in the managed resource, but that didn't work. got the error:
```
managed/iam.aws.upbound.io/v1beta1, kind=role  cannot run refresh: refresh failed: Invalid AWS Region: aws-global
```

do you guys think this is good enough and we don't need the providerconfig option? so basically any region-less managed resource will use aws-global for STS. that seems simple enough to me

Reply:
This seems reasonable to me -- we can always expand the API of `ProviderConfig` if needed. So in summary:
- If an assume role chain is being used and the MR specifies a region, we will use that region endpoint for the STS operations.
- If chain is being used and the MR does not specify a region, we use the global STS endpoint.

I think potentially using a separate region's STS endpoint could be desired by a user, but let's wait until it is requested since it involves API expansion. Are you aligned @ulucinar?

Reply:
I don't much about the use cases here. But one hypothetical scenario I may imagine is what happens if networking access to the host `sts.amazonaws.com` is blocked (due to law, temporary issue, etc.). But my understanding is that it (role chaining) was not working up to now for region-less resources and if we make a backward-compatible change (i.e., only revert back to `sts.amazonaws.com` for role chaining if the MR does not specify a region), we should not be breaking any existing (already working) scenarios and would be solving the problem for at least a subset of the users (of provider-aws). 

So, I think we are good to go. But to be on the safe side, even if, currently, we will not allow configuration of the STS region for region-less resources, let's scope the change only for region-less resources (so that the change is backward-compatible as discussed above).

Reply:
ok I pushed the change. it only sets the region to aws-global for STS requests when no region is set, so it should only occur in situations where it currently doesn't work at all